lib/toaster/toastergui/api.py

```diff
@@ -0,0 +1,1212 @@
+#
+# BitBake Toaster Implementation
+#
+# Copyright (C) 2016        Intel Corporation
+#
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Please run flake8 on this file before sending patches
+
+import os
+import re
+import logging
+import json
+import subprocess
+from collections import Counter
+
+from orm.models import Project, ProjectTarget, Build, Layer_Version
+from orm.models import LayerVersionDependency, LayerSource, ProjectLayer
+from orm.models import Recipe, CustomImageRecipe, CustomImagePackage
+from orm.models import Layer, Target, Package, Package_Dependency
+from orm.models import ProjectVariable
+from bldcontrol.models import BuildRequest, BuildEnvironment
+from bldcontrol import bbcontroller
+
+from django.http import HttpResponse, JsonResponse
+from django.views.generic import View
+from django.urls import reverse
+from django.db.models import Q, F
+from django.db import Error
+from toastergui.templatetags.projecttags import filtered_filesizeformat
+
+# development/debugging support
+verbose = 2
+def _log(msg):
+    if 1 == verbose:
+        print(msg)
+    elif 2 == verbose:
+        f1=open('/tmp/toaster.log', 'a')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you&#39;re creating or opening a file in global directory. Manipulating files from these directories can expose your application to race conditions. Also, an attacker can access, modify, corrupt, or delete files from these directories. To prevent this, use a dedicated sub-folder with controlled permission or secure-by-design APIs to create temporary files. For more information, see [CWE-379](https://cwe.mitre.org/data/definitions/379.html)